app/src/main/java/com/example/task3application/guess/GuessFragment.kt

```diff
@@ -24,6 +33,77 @@ class GuessFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        setListeners()
+        setObserve()
+    }
+
+    private fun setObserve() {
+        observe(viewModel.char) {
+            with(binding) {
+                tvChar.visibility = View.VISIBLE
+                tvChar.text = it.toString()
+                randomChar = it
+            }
+        }
+    }
+
+    private fun setListeners() {
+        with(binding) {
+            btnReset.setOnClickListener {
+                viewModel.resetGame()
+                tvInfo.text = getString(R.string.failure)
+            }
+            btnGuess.setOnClickListener {
+                if (viewModel.checkGame(randomNumber)) {
+                    tvInfo.text = getString(R.string.success)
+                    tvChar.text = randomNumber.toString()
+                    sharedViewModel.updateChar(randomChar)
+                    navigate(GuessFragmentDirections.actionGuessFragmentToDetailFragment())
+                } else {
+                    tvInfo.text = getString(R.string.failure)
+                }
+            }
+            btnZero.setOnClickListener {
+                randomNumber = 0
+                tvInfo.text = getString(R.string.zero)
+            }
+            btnOne.setOnClickListener {
```

Comment:
Küçük bir ödev, dokuz tane button var. Her birisi için ayrı ayrı `setOnClickListener` ve `randomNumber` atamasını yapıyoruz. Bu sayı daha da fazla olabilirdi. Bu işi daha kısa bir şekilde yazabilir miyiz, nasıl yapabiliriz, araştırma fırsatın oldu mu?.